components/dialog/test/dialog-ifrau.visual-diff.js

```diff
@@ -15,19 +15,59 @@ describe('d2l-dialog-ifrau', () => {
 
 	after(async() => await browser.close());
 
+	const getDialog = dialogContainer => {
+		const iframe = dialogContainer.querySelector('iframe');
+		return iframe.contentWindow.document.body.querySelector('#ifrau-dialog');
+	};
+
+	const open = async(page, selector) => {
+		const openEvent = oneEvent(page, selector, 'ifrau-dialog-open');
+		await page.$eval(selector, dialogContainer => {
+			const dialog = getDialog(dialogContainer);
+			dialog.opened = true;
+		});
+		return openEvent;
+	};
+
+	const reset = async(page, selector) => {
+		const closeEvent = oneEvent(page, selector, 'ifrau-dialog-close');
+		await Promise.race([
+			page.$eval(selector, dialogContainer => {
+				return new Promise(resolve => {
+					const dialog = getDialog(dialogContainer);
+					
+					dialog._fullscreenWithin = 0;
+					dialog.shadowRoot.querySelector('.d2l-dialog-content').scrollTo(0, 0);
+					if (dialog._state) {
+						dialog.opened = false;
+					} else {
+						resolve();
+					}
+				});
+			}),
+			closeEvent
+		]);
+
+		return page.$eval(selector, dialogContainer => {
+			const iframe = dialogContainer.querySelector('iframe');
+			iframe.contentWindow.document.body.querySelector('#open').click();
+		});
+	};
```

Comment:
I'm not 100% sure these tests will pass. They worked locally, but I also don't exactly know what the results were supposed to look like and it's hard to compare locally vs. CI by just looking at static images. I'll let CI tell me whether I need to fix these further. :)

... Also, this repetition of the `open` and `reset` stuff is kind of gross, but the existing helpers don't quite cut it because there's no single selector that can peek into the iframe contents.